R/calc_survival_probability_merton.R

```diff
@@ -0,0 +1,48 @@
+#' Calculate survival probability
+#'
+#' Function calculates survival probability for a maturity based on a structural
+#' Merton model.
+#' For details on implementation please compare [CreditRisk::Merton()].
+#' Unlike [CreditRisk::Merton()] this implementation:
+#' 1. only holds functionality to calculate probability of survival
+#' 1. can be called in vectorised fashion
+#' 1. has additional input validation by checking that all input values are
+#' non negative numeric vectors of the same length.
+#'
+#' @param L Numeric vector, holding dept values at maturity.
+#' @param V0 Numeric vector, holding values at time t0.
+#' @param sigma Numeric vector, holding volatility values.
+#' @param r Numeric vector, holding risk free rates.
```

Comment:
risk free interest rate

Reply:
just to make sure, in the original description it says risk free rate without interest, are those two redunant?